11-if/02-if-statement/05-challenge-userpass/02-2nd-challenge/01-challenge/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	usage     = "Usage: [username] [password]"
+	errUser   = "Access denied for %q.\n"
+	errPwd    = "Invalid password for %q.\n"
+	accessOK  = "Access granted to %q.\n"
+	user      = "jack"
+	pass      = "1888"
+	user2     = "inanc"
+	password2 = "1879"
+)
+
+type CheckAssessResult int
+
+const (
+	Allowed CheckAssessResult = 1 + iota
+	InvalidPassword
+	AccessDenied
+)
+
+func checkAccess(username, password string, desiredUsername, desiredPassword string) CheckAssessResult {
+	if username == desiredUsername && password == desiredPassword {
+		return Allowed
+	} else if password != desiredPassword {
+		return InvalidPassword
+	} else {
```

Comment:
you could ommit else here and put return at the end of function
so instead of this part
```
	} else {
		return AccessDenied
	}
```
you do it like this
```
	}
	return AccessDenied
```